test/RemoteAST/existentials.swift

```diff
@@ -0,0 +1,18 @@
+// RUN: %target-swift-remoteast-test %s | %FileCheck %s
+
+// REQUIRES: swift-remoteast-test
+
+@_silgen_name("printDynamicTypeAndAddressForExistential")
+func printDynamicTypeAndAddressForExistential(_: Any,
+                                              forType : Any.Type)
```

Comment:
So, this will only let you test that you handle various cases of types embedded within opaque existentials, because you're passing in an `Any`.  If you want to handle other existential types — `Error`, class-bounded existentials, existentials with actual bounds — you should use a generic function:

``func printDynamicTypeAndAddressForExistential<T>(_: T)``

That should still be passed through as a `void*` and a `const Metadata*`.  The trick is that it will be the caller's responsibility to actually pass in an existential value (e.g. `small_struct as Any`), because of course otherwise `T` will just be `MyStruct` or whatever.